### PR TITLE
feat: Adding a Pull Request Template

### LINKED ISSUE #200: Create a pull-request template
### Discussed in https://github.com/TimefoldAI/timefold-solver/discussions/502

<div type='discussions-op-text'>

<sup>Originally posted by **zepfred** December 18, 2023</sup>
The goal is to create a standardized process for updating quickstart modules and a pull request (PR) template that includes checkbox fields that must be marked before submitting or merging a PR. Here are some examples:

1. Does the proposed change, if needed, incorporate tests to prevent any regressions?
2. Are the UI and JS files fully tested?
    - Did you check all UI buttons (E.g., solve and analyze buttons)? 
    - Did you check the network calls?
    - Did you check the console messages?
3. Do you have a green build?</div>

### PR DESCRIPTION
**Pull request by zepfred**
## Description of the change

This pull request comes with a new template for submitting PRs and a checklist.

### Development

- [x] The changes have been covered with tests, if necessary.
- [x] You have a green build, with the exception of the flaky tests.
- [x] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [x] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [x] The console messages are validated for all modules affected by your changes.

### Code Review

- [x] This pull request includes an explanatory title and description.
- [x] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+## Description of the change
+
+> Description here
+
+## Related Issue
+
+Issue #ID
```
Comment (triceo) [inline .github/pull_request_template.md]: If you take a look on the right, you will see that Github allows linking a Github issue. I think this should be enough and we don't need to include it in the PR description as well.

WDYT?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Checklist
+
+### Development
+
+- [ ] The changes have been covered with tests, if necessary.
+- [ ] You have a green build, with the exception of the flaky tests.
+- [ ] UI and JS files are fully tested, and the UI buttons work for all modules or those affected by your changes (e.g., solve and analyze buttons).
```
Comment (triceo) [inline .github/pull_request_template.md]:
```suggestion
- [ ] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] The network calls work for all modules or those affected by your changes (e.g., solving a problem).
```
Comment (triceo) [inline .github/pull_request_template.md]:
```suggestion
- [ ] The network calls work for all modules affected by your changes (e.g., solving a problem).
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] The console messages are validated for all modules or those affected by your changes.
```
Comment (triceo) [inline .github/pull_request_template.md]:
```suggestion
- [ ] The console messages are validated for all modules affected by your changes.
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+### Code Review
+
+- [ ] This pull request includes an explicative title and description.
```
Comment (triceo) [inline .github/pull_request_template.md]:
```suggestion
- [ ] This pull request includes an explanatory title and description.
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] The GitHub issue is linked.
+
+### Merging
+
```
Comment (triceo) [inline .github/pull_request_template.md]: I'd remove the header. Merging will happen if code review is OK.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] At least one other engineer has approved the changes.
+- [ ] After merging the change, the issue requires you to follow up with the user.
```
Comment (triceo) [inline .github/pull_request_template.md]: What do you mean by this last point?

**Review by triceo — COMMENTED** [review]
Thanks for looking into this!
I've made some suggestions and asked for clarification here and there.

File: .github/pull_request_template.md (hunk 1 above)
Comment (zepfred) [reply .github/pull_request_template.md]: I've been working with GH for so long and never noted this field. Okay, we don't need to include this in the template.

File: .github/pull_request_template.md (hunk 7 above)
Comment (zepfred) [reply .github/pull_request_template.md]: The goal is to remind the owner to follow up with the user who opened the Issue. Let's say a user/customer finds a bug and files an issue. We should get back to him once the bug is fixed.

Comment (triceo) [reply .github/pull_request_template.md]: In that case, I suggest to rephrase a bit: "After PR is merged, inform the reporter."

Comment (zepfred) [reply .github/pull_request_template.md]: Done!

**Review by triceo — APPROVED** [review]

**Pull request merged**